controller/internal/enforcer/applicationproxy/applicationproxy.go

```diff
@@ -66,7 +66,8 @@ func NewAppProxy(tp tokenaccessor.TokenAccessor, c collector.EventCollector, puF
 		return nil, err
 	}
 
-	if ok := systemPool.AppendCertsFromPEM(s.PublicSecrets().CertAuthority()); !ok {
+	// We append the CA only if we are not in PSK mode as it doesn't provide a CA.
+	if ok := systemPool.AppendCertsFromPEM(s.PublicSecrets().CertAuthority()); !ok && (s.PublicSecrets().SecretsType() != secrets.PSKType) {
 		return nil, fmt.Errorf("Cannot append ca chain")
 	}
 
```

Comment:
Don't return an error here. If there is no CA, we just use the system pool. Just check if the secrets type is not PSK and ignore it if it is.

Reply:
This should be what this code is doing. I will  change it to make it more clear.